rosbag2_storage_default_plugins/test/rosbag2_storage_default_plugins/sqlite/test_sqlite_storage.cpp

```diff
@@ -322,6 +322,28 @@ TEST_F(StorageTestFixture, loads_config_file) {
       rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE));
 }
 
+TEST_F(StorageTestFixture, resilient_storage_writing_applies_over_configuration) {
+  // Check that "resilient" values are applied
+  const auto journal_setting = "\"journal_mode = some_value\"";
+  const auto synchronous_setting = "\"synchronous = another_value\"";
+  const auto not_overriden_setting = "\"cache_size = 1337\"";
+  const auto to_be_overriden_yaml = std::string("write:\n  pragmas: [") +
+    journal_setting + ", " + synchronous_setting + ", " + not_overriden_setting + "]\n";
+  const auto writable_storage = std::make_unique<rosbag2_storage_plugins::SqliteStorage>();
+  auto options = make_storage_options_with_config(to_be_overriden_yaml, kPluginID);
+  options.resilient_storage_writing = true;
+  writable_storage->open(options, rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE);
+
+  // resilient_storage_writing should replace "some_value" with "wal
+  EXPECT_EQ(writable_storage->get_storage_setting("journal_mode"), "wal");
+
+  // resilient_storage_writing should replace "another_value" with "2"
```

Comment:
Comment does not match code. The comment mentions setting "another_value" to "2", while the code is checking against "1".

Reply:
The correct value for default is NORMAL (returned as 1), so the comment is mistaken (commented before I checked) and the code is right.

Fixed

https://www.sqlite.org/pragma.html#pragma_synchronous